python/ShapeTools.py

```diff
@@ -91,10 +142,13 @@ def doIndividualModels(self):
                 # now we multiply by all the nuisances, but avoiding nested products
                 # so we first make a list of all nuisances plus the RooAddPdf
                 sumPlusNuis_s = ROOT.RooArgList(self.out.nuisPdfs); sumPlusNuis_s.add(sum_s)
+                pdf_bins = ROOT.RooProdPdf('pdfbins_bin%s' % b, '', binconstraints)
```

Comment:
Also add the bin-by-bin constraint pdfs to the channel pdf. Note that only the constraints for given channel are added at the moment. This is fine for CMS only use since combine scans the whole SimultaneousPdf to extract the nuisance constraints, but when we share with ATLAS this may not work.